common/persistence/data_manager_interfaces_test.go

```diff
@@ -232,3 +233,279 @@ func TestTimeStampConvertion(t *testing.T) {
+				"key2": []byte{},
+			},
+		},
+		{
+			name: "TC5: Memo contains duplicate byte slices",
+			inputMemo: map[string][]byte{
+				"key1": []byte("dup"),
+				"key2": []byte("dup"),
+			},
+			expectedOutput: map[string][]byte{
+				"key1": []byte("dup"),
+				"key2": []byte("dup"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			execInfo := &WorkflowExecutionInfo{
+				Memo: tt.inputMemo,
+			}
+
+			copyMemo := execInfo.CopyMemo()
+
+			// Check if both are nil
+			if tt.expectedOutput == nil {
+				if copyMemo != nil {
+					t.Errorf("Expected nil, got %v", copyMemo)
+				}
+				return
+			}
+
+			// Check if lengths match
+			if len(copyMemo) != len(tt.expectedOutput) {
+				t.Errorf("Expected map length %d, got %d", len(tt.expectedOutput), len(copyMemo))
+			}
+
+			// Check each key-value pair
+			for key, expectedVal := range tt.expectedOutput {
+				copiedVal, exists := copyMemo[key]
+				if !exists {
+					t.Errorf("Expected key %s not found in copy", key)
+					continue
+				}
+
+				if !bytes.Equal(copiedVal, expectedVal) {
+					t.Errorf("For key %s, expected value %v, got %v", key, expectedVal, copiedVal)
+				}
+
+				// Ensure that the byte slices are different underlying arrays (deep copy)
+				if len(copiedVal) > 0 && &copiedVal[0] == &expectedVal[0] {
+					t.Errorf("For key %s, byte slices reference the same underlying array", key)
+				}
+			}
+		})
+	}
+}
+
+func TestCopySearchAttributes(t *testing.T) {
+	tests := []struct {
+		name                string
+		inputSearchAttrs    map[string][]byte
+		expectedOutputAttrs map[string][]byte
+	}{
+		{
+			name:                "TC1: SearchAttributes is nil",
+			inputSearchAttrs:    nil,
+			expectedOutputAttrs: nil,
+		},
+		{
+			name:                "TC2: SearchAttributes is empty",
+			inputSearchAttrs:    map[string][]byte{},
+			expectedOutputAttrs: map[string][]byte{},
+		},
+		{
+			name: "TC3: SearchAttributes contains multiple entries",
+			inputSearchAttrs: map[string][]byte{
+				"attr1": []byte("value1"),
+				"attr2": []byte("value2"),
+			},
+			expectedOutputAttrs: map[string][]byte{
+				"attr1": []byte("value1"),
+				"attr2": []byte("value2"),
+			},
+		},
+		{
+			name: "TC4: SearchAttributes contains empty byte slices",
+			inputSearchAttrs: map[string][]byte{
+				"attr1": []byte(""),
+				"attr2": []byte{},
+			},
+			expectedOutputAttrs: map[string][]byte{
+				"attr1": []byte(""),
+				"attr2": []byte{},
+			},
+		},
+		{
+			name: "TC5: SearchAttributes contains duplicate byte slices",
+			inputSearchAttrs: map[string][]byte{
+				"attr1": []byte("dup"),
+				"attr2": []byte("dup"),
+			},
+			expectedOutputAttrs: map[string][]byte{
+				"attr1": []byte("dup"),
+				"attr2": []byte("dup"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			execInfo := &WorkflowExecutionInfo{
+				SearchAttributes: tt.inputSearchAttrs,
+			}
+
+			copyAttrs := execInfo.CopySearchAttributes()
```

Comment:
also validate that changing execInfo.SearchAttributes doesn't change `copyAttrs` (ensure it was copoied). let's do this in other tests as well

Reply:
copy is already ensured by comparing the byte slices